internal/resource/common/id/id_schema.go

```diff
@@ -0,0 +1,18 @@
+package id
+
+import (
+	datasourceschema "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+func DataSourceSchema(s *datasourceschema.Schema, required bool, description string) {
```

Comment:
I'd rename this to something like AddToDataSourceSchema to imply that the schema object passed in is being modified by this method